samples/fpgaperf_counter/fpgaperf_counter.c

```diff
@@ -558,6 +567,10 @@ fpga_result fpgaPerfCounterStartRecord(fpga_perf_counter *fpga_perf)
 		return FPGA_INVALID_PARAM;
 	}
 
+	if (!fpga_perf->previlege) {
```

Comment:
Then, instead of checking here and returning FPGA_OK (it should technically be an error the way we are defining it.), we would remove these checks and rely on the check in host_exerciser before the perf counters are queried.